File: old_frontend/apiClient.py
import requests
import os

def get_api_status():
    url = "https://esva-demo-soel3q6bbq-uw.a.run.app"+ "/test"
    response = requests.get(url)
    if response.status_code == 200:
        return response.json()['message']
    else:
        return 'Error al obtener la respuesta de la API status'
    
def login(username, password):
    url = "https://esva-demo-soel3q6bbq-uw.a.run.app" + "/api/login"
    data = {'username': username, 'password': password }
    response = requests.post(url, json=data)
    if response.status_code == 200:
        # Verificar si la respuesta contiene el token de portador
        bearer_token = response.text.strip()  # Eliminar espacios en blanco alrededor del token
        if bearer_token:
            return bearer_token
        else:
            return 'error'
    else:
        return 'connection error'

def validate_token(token):
    url = "https://esva-demo-soel3q6bbq-uw.a.run.app"+ "/api/verify/token"
    headers = {
        "Authorization": f"Bearer {token}"
    }
    response = requests.get(url, headers=headers)
    if response.status_code == 200:
        return response.json()
    else:
        return 'Error al obtener la respuesta de la API validate token'
    
def ask_question(query, token):
    url = "https://esva-demo-soel3q6bbq-uw.a.run.app"+ "/api/getAnswer"
    data = {'query': query}
    headers = {
        "Authorization": f"Bearer {token}"
    }
    response = requests.post(url, json=data, headers=headers)
    if response.status_code == 200:
        return response.json()['message']
    else:
        return 'Error al obtener la respuesta de la API'

    

def main():
    # prueba de consultas de apis
    api_status = get_api_status()
    print(f"resultado de api_status: {api_status}")
    bearer_token = login(username="Admin@west.net.co", password="West2024")
    print(f"resultado del bearer token {bearer_token}")
    data_token = validate_token(bearer_token)
    print(f"resultado del data token {data_token}")
    response = ask_question("cuales son las causas para un despido?", bearer_token)
    print(f"respuesta a la pregunta {response}")

if __name__ == "__main__":
    main()

File: frontend/apiClient.js
    // Script.js

    let tokenJWT;


    function test() {
        const url = 'https://esva-demo-soel3q6bbq-uw.a.run.app/test';
        
    
        fetch(url)
            .then(response => {
                if (!response.ok) {
                    throw new Error('Error en la solicitud');
                }
                return response.json();
            })
            .then(data => {
                console.log('Respuesta de la API sin Token:', data);
                // Aquí puedes manejar la respuesta de la API como desees
            })
            .catch(error => {
                console.error('Error al consultar la API sin Token:', error);
            });
    }

    function login() {
        const url = 'https://esva-demo-soel3q6bbq-uw.a.run.app/api/login';
        const username = document.getElementById('username').value;
        const password = document.getElementById('password').value; 
        const credentials = {
            username: username,
            password: password
        };

        const options = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(credentials)
        };
        // console.log(url, options)
        fetch(url, options)
            .then(response => {
                if (!response.ok) {
                    throw new Error('Error en la solicitud');
                }
                return response.text(); // Obtener el texto de la respuesta
            })
            .then(text => {
                tokenJWT = text.trim(); // Asignar el token a la variable global, eliminando espacios en blanco
                document.getElementById("loginContainer").style.display = "none";

                // console.log('Token JWT:', tokenJWT);
                // Aquí puedes manejar el token JWT como desees
            })
            .catch(error => {
                console.error('Error al consultar la API sin Token:', error);
            });
    }

    function imprimirJWT() {
        console.log(tokenJWT)
    }

    function testToken() {
        const url = 'https://esva-demo-soel3q6bbq-uw.a.run.app/api/verify/token';

        const options = {
            method: 'GET',
            headers: {
                'Authorization': `Bearer ${tokenJWT}`
            }
        };

        fetch(url, options)
            .then(response => {
                if (!response.ok) {
                    throw new Error('Error en la solicitud');
                }
                return response.json();
            })
            .then(data => {
                console.log('Respuesta de la API con Token:', data);
                // Aquí puedes manejar la respuesta de la API como desees
            })
            .catch(error => {
                console.error('Error al consultar la API con Token:', error);
            });
    }

    function makeQuestion(query) {
        const url = 'https://esva-demo-soel3q6bbq-uw.a.run.app/api/getAnswer';
                
        const credentials = {
            query: query
        };

        const options = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${tokenJWT}`
            },
            body: JSON.stringify(credentials)
        };

        fetch(url, options)
            .then(response => {
                if (!response.ok) {
                    throw new Error('Error en la solicitud');
                }
                return response.json();
            })
            .then(data => {
                console.log('Respuesta de la API con Token:', data);
                // Aquí puedes manejar la respuesta de la API como desees
            })
            .catch(error => {
                console.error('Error al consumir la API con Token:', error);
            });
    }
